### ISSUE TITLE
handle koji longform seconary arch stuff better

### ISSUE THREAD

**Issue by ralphbean** [fedora-infra/fedmsg_meta_fedora_infrastructure#243]
Some useful comments from irc:

```
 dgilmore │ threebean: I know what that fmn issue is
 dgilmore │ threebean: secondary arches directly import noarch only builds.
 dgilmore │ threebean: so there is no host
 dgilmore │ as there was no task to build it
 dgilmore │ threebean: there is no task id for those builds either
 dgilmore │ threebean: so "info = sess.getTaskInfo(taskid)" can not work
 dgilmore │ as there is no taskid
 dgilmore │ threebean: actually none of it should work
 dgilmore │ threebean: there also likely should not be a notification for it
 dgilmore │ threebean: http://paste.fedoraproject.org/204899/74073514/ see the output of getBuild
 dgilmore │ task_id == None
threebean │ secondary arch koji stuff is turned off for everyone by default.. but there's a couple people who want it and have it turned o
threebean │ dgilmore: those details help.  I'll file them away in the ticket for when I can circle back on it.
threebean │ thanks for digging into it :)
 dgilmore │ threebean: sure, but importing a noarch build should probably be skipped
        * │ threebean nods
 dgilmore │ threebean: at the least you should check if task_id is None and skip trying to get the details
threebean │ sounds good
 dgilmore │ and likely print something along the lines of imported noarch build from primary
```

**Comment by ralphbean** [comment]
Patch from dennis gilmore:

```diff
This is normally due to importing a noarch only build on secondary
arches from primary. if it happens on primary it is likely an error.
fixes https://github.com/fedora-infra/fedmsg_meta_fedora_infrastructure/issues/243

Signed-off-by: Dennis Gilmore <dennis@ausil.us>
---
 fedmsg_meta_fedora_infrastructure/buildsys.py | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/fedmsg_meta_fedora_infrastructure/buildsys.py b/fedmsg_meta_fedora_infrastructure/buildsys.py
index 8cf105f..54e7514 100644
--- a/fedmsg_meta_fedora_infrastructure/buildsys.py
+++ b/fedmsg_meta_fedora_infrastructure/buildsys.py
@@ -121,12 +121,15 @@ class KojiProcessor(BaseProcessor):
             log.warning(unicode(e))
             _build_str = unicode(e) + "\n"
 
-        try:
-            _task_str = "Closed tasks:\n-------------\n"
-            _task_str += cls._fill_task_template(sess, full_build['task_id'])
-        except Exception as e:
-            log.warning(unicode(e))
-            _task_str = unicode(e) + "\n"
+        if full_build['task_id'] is None:
+            _task_str = "Build imported into koji\n"
+        else:
+            try:
+                _task_str = "Closed tasks:\n-------------\n"
+                _task_str += cls._fill_task_template(sess, full_build['task_id'])
+            except Exception as e:
+                log.warning(unicode(e))
+                _task_str = unicode(e) + "\n"
 
         return _build_str + _task_str
 
-- 
2.3.2
```

**Comment by ralphbean** [comment]
Fixed in 1b0a0065924fcfbfd71ef2c1e8dfa17269cd44bb
